docs/decisions/0003-lightweight-extension-points.rst

```diff
@@ -0,0 +1,202 @@
+Lightweight extension points
+############################
+
+Status
+******
+
+**Draft** *2024-02-27*
+
+Context
+*******
+
+Open-ended questions are commonly used in education for assessment purposes, but their flexible nature can facilitate cheating. Therefore, instructors need a way of assessing when cheating happens when reviewing responses. One widely-used solution to this problem in the academic field is the use online plagiarism tools by instructors. These tools receive the students' responses, analyze them, and provide reports that assist instructors in scoring.
+
+No standard exists for ORA to facilitate integration with tools that process students' answers for plagiarism or other useful reports for the course staff. Therefore, ORA must allow external interactions with students' responses so they can be sent for analysis.
+
+Decisions
+*********
+
+We'll enable external interactions with students' responses by using the `Hooks Extensions Framework`_. Using this approach, we'll implement extension points with minimal modifications in the edx-ora2 repository, making interventions with the students' response lifecycle possible. This is an interim solution for the problem stated above while the community develops a more sophisticated process that covers more use cases. 
+
+This approach includes implementing two extension points, which definitions will live in `openedx-filters`_ and `openedx-events`_ accordingly:
+
+For the first extension point we will use an `Open edX Filter`_ with the following definition:
+
+
+..  code-block:: python
```

Comment:
To make the ADR more readable I suggest we move the code to a second file with code examples that we can reference from this text, but keep the conceptual part as the center of this file.

Reply:
I was researching a bit about where the code for the proposal should live and found this: 

https://docs.openedx.org/projects/openedx-proposals/en/latest/best-practices/oep-0019-bp-developer-documentation.html#adrs

> Architecture Decision Records (ADRs) are local lightweight documents of technical decisions co-located with their corresponding code...

So that's why I'm trying to illustrate what this will look like, but in reality, we should use the implementation as examples. I propose removing the code altogether without adding a new file, and then, if needed, we add references to the actual code implemented. What do you think?